src/Databases/DatabaseOnDisk.cpp

```diff
@@ -644,19 +646,24 @@ void DatabaseOnDisk::iterateMetadataFiles(ContextPtr local_context, const Iterat
             throw Exception(ErrorCodes::INCORRECT_FILE_NAME, "Incorrect file extension: {} in metadata directory {}", file_name, getMetadataPath());
     }
 
+    /// Don't wait forever for threads from the Global thread pool. Otherwise, ClickHouse can enter a deadlocked state
+    /// where it blocks infinitely on startup waiting for threads.
+    const std::chrono::microseconds lock_acquire_timeout_microseconds = std::chrono::seconds{getContext()->getSettingsRef().lock_acquire_timeout};
```

Comment:
lets just do
`getContext()->getSettingsRef().lock_acquire_timeout.totalMicroseconds();`

it is a bit tangle how it is converted to std::chrono. `getContext()->getSettingsRef().lock_acquire_timeout` has a type as `SettingFieldTimespan`.